zilliqa/src/consensus.rs

```diff
@@ -554,7 +559,14 @@ impl Consensus {
                     % self.config.consensus.consensus_timeout.as_secs())
                     == 0
             {
-                return Ok(Some(self.build_new_view()?));
+                if let Some((_, ExternalMessage::NewView(new_view))) =
+                    self.new_view_message_cache.as_ref()
+                {
+                    if new_view.view == self.get_view()? {
+                        return Ok(self.new_view_message_cache.clone());
+                    }
+                }
+                self.new_view_message_cache = None;
```

Comment:
If we clear this straight away, won't we only ever send a single repeated `NewView`?

Reply:
Maybe a better check is something like `if self.view == self.new_view_message_cache.1.view`? (i.e. If I've sent and cached a `NewView` in this view already, then re-send it?

Reply:
That is what this code intends to do!

If were in the same view then `return` with the `NewView` message. If were not in the same view then clear the cache.

Its more ugly than `self.new_view_message_cache.1.view` to avoid unwrapping of Options

Reply:
I've moved the clear cache line inside of the `if let` to be safer.